### ISSUE TITLE
Is training possible when all the images in dataset have same size?

### ISSUE THREAD

**Issue by YangJae96** [lmy98129/VLPD#4]
Hi.

Does your code only work on fixed size input images?

Did you set the input size to 640x1280 in config.py since all the images in Cityperson dataset is 640x1280?

What if there are different size of images in one dataset? Is it still possible to train the model with your code?

Thank you.

**Comment by lmy98129** [comment]
1. Currently, we did not evaluate our code on dataset with different sizes, e.g., CrowdHuman. 

- Because both Citypersons and Caltech use videos captured from the camera of urban driving vehicles, which have fixed size of video frames. 
- Our proposed VLPD is not designed for intra-class occlusion (i.e., pedestrian occludes each other). Known as a "context-aware" method, it focuses on context-related problems.
- For example, inter-class occlusion between pedestrians and contextual objects, the ambiguity of far and blur small pedestrians and objects, and human-like objects (traffic signs, trees, etc.) in nearer scenes, are visualized in the supplementary materials and evaluated by various subsets. 
- Therefore, even VLPD can be trained on datasets for crowded scenes, its performance will be not guaranteed. 

2. Meanwhile, some previous methods for crowd scene dataset like CrowdHuman also need a fixed image resizing.

- The PyTorch dataloader only supports fixed image size for parallelization, i.e., images in a mini-batch should be resized. 
- Therefore, previous method like CrowdDet also has to perform a fixed image resizing. Please refer to the code below: https://github.com/Purkialo/CrowdDet/blob/4a0c674c40fdcb3e2706d39544f088390f9f63fe/model/retina_fpn_baseline/config.py#L31

By the way, the image size is one of the FAQs (Frequently-Asked Questions) in our received emails, thus we provide a detailed summary of the previous replies. Please inform us if there are more problems. 

Mengyin Liu

**Comment by YangJae96** [comment]
@lmy98129  
Thanks for the reply.
I was hoping to apply your method in other pedestrian detection dataset that are not crowded.
If the above issue could be solved, it would be available to extend it to other downstream tasks that needs to perform pedestrian detection first.

1. If I want to train your method with the dataset that contains images with various size,
should I set the mix_size and max_size of the width, height of the image?
For example, if the dataset contains images with two sizes -> w,h   (1920,1080), (1500,900),
should I resize the image of (1500,900) to (1920,1080) by zero padding to make
images in the mini-batch all same before feeding it to the model?

2. Since your code estimate GT for offset center and scale in dataloader
https://github.com/lmy98129/VLPD/blob/main/dataloader/loader.py#97,
if I want to perform fixed image resizing like in CrowdDet, should the above GT should also be changed according to the resized images?


2. Also, how should I access the FAQs in your email? I want to know more about the details of the image sizes.

**Comment by lmy98129** [comment]
1. Yes, the same image size in a mini-batch is necessary for the PyTorch dataloader. The implementation of previous method like CrowdDet can be a reference. Possible approaches are diversified, e.g., resizing and padding.
 
2. GT should be consistent with the original input image size. However, after checking the code, we notice that "cfg.size_train" does not control the original input image size, but **_the image size after the augmentation_**. 

- Please refer to the [code](https://github.com/lmy98129/VLPD/blob/main/dataloader/data_augment.py#L139) in data augmentation where the "cfg.size_train" is used for random cropping and random paving after the random resizing. 

- So it means the original image size **_can be different_**. But the images after augmentation are always kept the same, not matter which image size is used. 

- Then augmented images are [feed into](https://github.com/lmy98129/VLPD/blob/main/dataloader/loader.py#L92) the Pytorch Dataloader (actually the Dataset of the Dataloader).  

- **_Recommendation 1_**: Just set a suitable "cfg.size_train" (larger one might lead to more random paving, smaller one more random cropping) might make the training success, without any extra padding, resizing or modifying GT. 

- **_Recommendation 2_**: Meanwhile, "cfg.size_test" should be modified, which only accepts one image size of input image during testing (no augmentation). Simply no resizing during testing might be a solution. 

- Note that the code is inherited from our baseline CSP, which is also designed for Caltech and CityPersons datasets. Thus, we did not consider whether the code meets the requirements of multiple image sizes. 

3. FAQs does not mean a webpage, but my personal expression that image resizing is frequently asked by email, but it is the first time in the GitHub issues. The summary above is all the content of our previous replies, without omitting important details. 

To my best knowledge, the replies to other types of the problems can be found in the GitHub issues. Thank you for discussing with us about the image resizing, which inspires us to better understand such a new requirement. If you have more problems, just inform us. 

Mengyin Liu
